mautic/mautic
# Allow enabling adding new records and updating existing records in Pipedrive separately

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 



#### Description:


Previously, the setting labeled "Import contacts to Pipedrive immediately on change in Mautic" could only be toggled on or off regardless of operation.

This PR changes that setting into two options which can be enabled or disabled:
- Create immediately on create in Mautic
- Update immediately on update in Mautic

This adds the ability for Mautic to be configured to only create new entities in Pipedrive, or only update existing entities in Pipedrive, whenever that is a requirement by a particular use case.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Sign up for a test account in Pipedrive (or use an existing empty or test account)
3. Enable and configure the Pipedrive integration in Mautic to connect to your new Pipedrive account, and select both the "Create" and "Update" options mentioned earlier in the settings
4. Create a company and contact in Mautic and ensure that the records both exist in Pipedrive now
5. Update both of the new records with new data in Mautic, and ensure that the corresponding data is updated in Pipedrive
6. Disable both the "Create" and "Update" options in the Pipedrive integration settings
7. Create a new company and contact and ensure that they do NOT get created in Pipedrive
8. Update the company and contact from step 4 with new data and ensure that it does NOT get updated in Pipedrive
9. Enable only the "Create" option without the "Update" option in the Pipedrive integration settings
10. Create a company and contact in Mautic and ensure that they DO exist in Pipedrive
11. Update the same records in Mautic and ensure that they DO NOT get updated in Pipedrive
12. Enable only the "Update" option without the "Create" option in the Pipedrive integration settings
13. Update the records created in step 4 and ensure that they DO get updated properly in Pipedrive
14. Create a company and contact in Mautic and ensure that they DO NOT get created in Pipedrive


It might be considered a backwards-compatibility break that a person has to edit the settings and reconfigure these two options after they update their existing site. If so, could this potentially be handled in a migration to keep it backwards-compatible?

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11230"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
